src/overlay/types.ts

```diff
@@ -1,4 +1,14 @@
-export interface OverlayProps {
+export type OverlayClickProps =
+    | {
+          enableOverlayClick?: false | undefined;
+          onOverlayClick?: (() => void) | undefined;
```

Comment:
shouldn't this be
```suggestion
          onOverlayClick?: undefined;
```
to prevent the callback from being set without the flag?

Reply:
if there ever is a use case of conditional enabling of `overlayClick`, typescript will complain when you pass in a function, since it expects undefined for the `false` case

Reply:
okay so this is just to ensure `onOverlayClick` is definitely set if `enableOverlayClick` is true?

Reply:
yup